### PR TITLE
refactor(lfs): improve API structure and add docs/tests

### PR DESCRIPTION
**Pull request by WiedersehenM**
## Overview

This PR completes the LFS API refactoring task by adding comprehensive unit tests and translating documentation to English.

## Related issue
#1691 

#### `mono/src/api/router/lfs_router.rs`

**Added:** Comprehensive unit test suite (16 test functions, ~280 lines)

**Test Coverage:**

- **Error Mapping Tests (7 tests):**
  - `test_map_lfs_error_not_found` - Tests 404 error mapping
  - `test_map_lfs_error_bad_request` - Tests 400 error mapping
  - `test_map_lfs_error_internal_server_error` - Tests 500 error mapping
  - `test_map_lfs_error_with_gitlfs_error` - Tests GitLFSError type handling
  - `test_map_lfs_error_edge_cases` - Tests edge cases (empty strings, case sensitivity)
  - `test_map_lfs_error_priority` - Tests error matching priority
  - `test_map_lfs_error_unicode_support` - Tests Unicode character support

- **Parameter Validation Tests (2 tests):**
  - `test_lfs_download_chunk_parameter_validation` - Tests chunk download parameter parsing
    - Valid parameters
    - Missing offset/size
    - Invalid format handling
  - `test_lfs_download_chunk_parameter_edge_cases` - Tests edge cases
    - Zero values
    - Large values (u64::MAX)
    - Negative numbers (should fail)

- **Response Format Tests (2 tests):**
  - `test_error_response_format` - Tests error response format consistency
  - `test_content_type_constants_consistency` - Tests Content-Type constants

- **Route Structure Tests (3 tests):**
  - `test_lfs_routes_structure` - Tests route creation
  - `test_lfs_router_paths` - Tests router path registration
  - `test_lfs_routes_all_endpoints` - Tests all endpoint registration

- **Data Structure Tests (2 tests):**
  - `test_request_object_default` - Tests RequestObject default values
  - `test_lfs_content_types` - Tests content type constants

**Total:** 16 unit tests covering error handling, parameter validation, edge cases, and route structure.

#### `docs/lfs-api.md`

**Changed:** Translated from Chinese to English

**Content:**

- Overview section
- Endpoints list (8 endpoints)
- Examples (Batch, Upload, Download, Lock Management)
- Developer Notes
- Related Files

#### `docs/lfs-api-improvements.md`

**Changed:** Translated from Chinese to English

**Content:**

- Comparison Baseline (GitHub official specification links)
- Issues Found and Fix Recommendations (7 issues)
- Priority levels (High/Medium/Low)

## Test Statistics

- **Total Tests Added:** 16
- **Test Categories:**
  - Error handling: 7 tests
  - Parameter validation: 2 tests
  - Response format: 2 tests
  - Route structure: 3 tests
  - Data structures: 2 tests

### Testing

-  Complete unit test coverage for LFS router layer
-  Error handling validation (404/400/500 status codes)
-  Parameter validation logic
-  Edge case handling
-  Route structure verification

### Documentation

-  Consistent English documentation matching other docs in `docs/` directory
-  Complete API usage examples
-  GitHub specification comparison documented

## Related Files

- Routes: `mono/src/api/router/lfs_router.rs`
- Business logic: `ceres/src/lfs/handler.rs`
- Data structures: `ceres/src/lfs/lfs_structs.rs`
- Documentation: `docs/lfs-api.md`, `docs/lfs-api-improvements.md`

## Notes

- All tests are unit tests that don't require external dependencies
- Tests follow the same pattern as other router tests in the codebase (e.g., `cl_router.rs`)
- Documentation style matches existing English documentation (e.g., `docs/api.md`)

### CODE REVIEW

File: mono/src/api/router/lfs_router.rs [hunk 1]
```diff
@@ -74,21 +94,39 @@ const LFS_CONTENT_TYPE: &str = "application/vnd.git-lfs+json";
...
-            .route("/locks", get(list_locks))
-            .route("/locks", post(create_lock))
-            .route("/locks/verify", post(list_locks_for_verification))
-            .route("/locks/{id}/unlock", post(delete_lock))
-            .route("/objects/batch", post(lfs_process_batch))
-            .route("/objects/{object_id}/chunks", get(lfs_fetch_chunk_ids)),
-    )
+    OpenApiRouter::new()
+        .nest("/info/lfs", lfs_routes())
+        .nest("/api/v1/lfs", lfs_routes())
```
Comment (benjamin-747) [inline mono/src/api/router/lfs_router.rs]: 这个请求路径的含义是什么？lfs 文档中好像并没有关于此路径的定义

File: mono/src/server/http_server.rs [hunk 2]
```diff
@@ -261,14 +261,48 @@ pub const MERGE_QUEUE_TAG: &str = "Merge Queue Management";
     tags(
```
Comment (benjamin-747) [inline mono/src/server/http_server.rs]: 这个写法不符合项目目前的规范，请移除260-305 行

File: ceres/src/lfs/handler.rs [hunk 3]
```diff
@@ -676,3 +676,86 @@ async fn lfs_check_sub_oid_exist(
     let result = storage.get_lfs_relations_ori_oid(sub_oid).await.unwrap();
     Ok(!result.is_empty())
 }
+
+#[cfg(test)]
+mod tests {
```
Comment (benjamin-747) [inline ceres/src/lfs/handler.rs:681]: 单元测试需要覆盖到LFS的具体逻辑，而不是仅测试输入输出

File: mono/src/api/router/lfs_router.rs [hunk 4]
```diff
@@ -74,21 +94,39 @@ const LFS_CONTENT_TYPE: &str = "application/vnd.git-lfs+json";
...
 }
 
+fn map_lfs_error<E: ToString>(err: E) -> (StatusCode, String) {
```
Comment (benjamin-747) [inline mono/src/api/router/lfs_router.rs]: 目前已经存在通用的错误处理逻辑，这个函数是否是必要的？

**Review by benjamin-747 — CHANGES_REQUESTED** [review]

**Comment by WiedersehenM** [comment]
好的老师，我刚刚同步了上游仓库的代码，我会针对您提到的问题对代码进行修改

File: mono/src/server/http_server.rs (hunk 2 above)
Comment (WiedersehenM) [reply mono/src/server/http_server.rs]: 已移除

File: mono/src/api/router/lfs_router.rs (hunk 1 above)
Comment (WiedersehenM) [reply mono/src/api/router/lfs_router.rs]: 两个路径使用相同的 handlers，功能完全一致，/info/lfs是标准的 Git LFS 协议路径，Git LFS 客户端能够对/info/lfs自动发现

File: mono/src/api/router/lfs_router.rs (hunk 4 above)
Comment (WiedersehenM) [reply mono/src/api/router/lfs_router.rs]: handler 层目前统一返回的是 `GitLFSError::GeneralError(String)`，仅有错误信息而没有**状态码**。为了更贴近Git LFS的开发文档，我用map_lfs_error通过字段匹配handler返回的错误信息，将报错进行了简单的分类并加上状态码。

**Comment by genedna** [comment]
@codex review

**Comment by genedna** [comment]
@codex review

**Comment by genedna** [comment]
@codex review

**Review by benjamin-747 — APPROVED** [review]

File: mono/src/server/http_server.rs [hunk 5]
```diff
@@ -1,498 +1,494 @@
...
+pub const LABEL_TAG: &str = "Label Management";
+pub const CONV_TAG: &str = "Conversation and Comment";
+pub const SYNC_NOTES_STATE_TAG: &str = "sync-notes-state";
+pub const USER_TAG: &str = "User Management";
+pub const REPO_TAG: &str = "Repo creation and synchronisation";
+pub const MERGE_QUEUE_TAG: &str = "Merge Queue Management";
+pub const BUCK_TAG: &str = "Buck Upload API";
+pub const LFS_TAG: &str = "Git LFS";
+#[derive(OpenApi)]
+#[openapi()]
```
[the remark this answers is not in the document]
Comment (WiedersehenM) [reply mono/src/server/http_server.rs:415]: 这个好像和当前项目的编码规范不符
